balancer/rls/control_channel.go

```diff
@@ -68,24 +72,39 @@ func newControlChannel(rlsServerName, serviceConfig string, rpcTimeout time.Dura
 		rpcTimeout:      rpcTimeout,
 		backToReadyFunc: backToReadyFunc,
 		throttler:       newAdaptiveThrottler(),
+		state:           buffer.NewUnbounded(),
 	}
 	ctrlCh.logger = internalgrpclog.NewPrefixLogger(logger, fmt.Sprintf("[rls-control-channel %p] ", ctrlCh))
 
 	dopts, err := ctrlCh.dialOpts(bOpts, serviceConfig)
 	if err != nil {
 		return nil, err
 	}
-	ctrlCh.cc, err = grpc.Dial(rlsServerName, dopts...)
+	ctrlCh.cc, err = grpc.NewClient(rlsServerName, dopts...)
 	if err != nil {
 		return nil, err
 	}
+	ctrlCh.unsubscribe = internal.SubscribeToConnectivityStateChanges.(func(cc *grpc.ClientConn, s grpcsync.Subscriber) func())(ctrlCh.cc, ctrlCh)
```

Comment:
Please add a comment here as to why we are subscribing to state updates before we start connecting on the channel.